front/src/Interpreter/Tokenizer/Tokenizer.ts

```diff
@@ -0,0 +1,184 @@
+import {literalType, Token} from "./Token.ts";
+import {TokenType} from "./TokenType.ts";
+import {keywords} from "./Keywords.ts"
+
+ 
+    
+export class Tokenizer {
+    private source: string;
+    private tokens: Token[] = []
+    private current: integer
+    private start: integer
+    
+    constructor(source: string) {
+        this.source = source;
+        this.current = 0
+        this.start = 0
+
+        this.deleteUnnecessarySpaces()
+    }
+
+    
+    scanTokens():Token[] {
+        while(!this.isAtEnd()) {
+            this.start = this.current
+            this.scanToken()
+        }
+
+        // this.tokens.push(new Token(TokenType.EOF, "", ));
+        return this.tokens
+    }
+
+    private deleteUnnecessarySpaces() {
+        this.deleteWhiteSpaces()
+        this.deleteComments()
```

Comment:
deleteComments method should be called before. as for comments to be deleted it must and with \n, but deleteWhiteSpaces removes them. this would result in emptying source